packages/controller/src/lib/notificationHandler.ts

```diff
@@ -305,21 +404,19 @@ class NotificationHandler {
     /**
      * Check the given message against all regular expressions of the given scope
      *
-     * @param {string} scope - scope of the message
-     * @param {string} message - message to check
-     * @return {any[]}
-     * @private
+     * @param scope - scope of the message
+     * @param message - message to check
      */
-    _parseText(scope, message) {
+    private _parseText(scope: string, message: string): string[] {
         const categories = [];
         if (this.setup[scope]) {
-            for (const category of Object.keys(this.setup[scope])) {
-                if (!this.setup[scope][category]) {
+            for (const category of Object.keys(this.setup[scope].categories)) {
+                if (!this.setup[scope].categories[category]) {
                     continue;
                 }
```

Comment:
```suggestion
            for (const [category, definition] of Object.entries(this.setup[scope].categories)) {
                if (!definition) continue;
```
and use `definition` below instead of repeating the indexed access.

How can `definition` be falsy anyways? Do we not check this when `this.setup[...]` is created?